apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopRightPanel.tsx

```diff
@@ -60,12 +63,36 @@ export function TlaEditorTopRightPanel({
 	)
 }
 
+function getFileName(editor: Editor) {
+	const documentName = editor.getDocumentSettings().name
+	if (documentName?.length > 0) return documentName
+	const firstPageName = editor.getPages()[0].name
+	if (firstPageName.length > 0 && !firstPageName.startsWith('Page 1')) return firstPageName
```

Comment:
Thought of using the localized copy here, but I think we have a long standing bug where we always use the 'Page 1' as the default name.